lib/Analysis/Allocation.cpp

```diff
@@ -88,30 +88,74 @@ getScratchConfigForCvtLayout(triton::gpu::ConvertLayoutOp op, unsigned &inVec,
   return paddedRepShape;
 }
 
-SmallVector<unsigned> getScratchConfigForReduce(triton::ReduceOp op) {
+/// ====================== reduce op config begin ==========================
+
+bool enableFastReduce(triton::ReduceOp op) {
+  auto srcTy = op.operand().getType().cast<RankedTensorType>();
```

Comment:
Nit: The input of reduce op should always be Tensor, but it might still be a good habit to add an assert here.

Reply:
I think `mlir::Type::cast()` method contains an assertion in its implementation.

Related links:

https://github.com/llvm/llvm-project/blob/main/mlir/include/mlir/IR/Types.h#L281

https://github.com/llvm/llvm-project/blob/main/llvm/include/llvm/Support/Casting.h#L564-L586